### ISSUE TITLE
Parameter being resolved multiple times

### ISSUE THREAD

**Issue by svandervlugt** [nelmio/alice#664]
I want to use a parameter having a unique id multiple times.
I define this parameter as follows:
```
parameters:
     unique_id: "<(uniqid())>"
```
And use this parameter in the following objects:
```
Hostnet\Task\Entity\FirstObject:
    first_object:
        identifier: "<{unique_id}>"
```
and 
```
Hostnet\Task\Entity\SecondObject:
    object:
        __construct:
            - "@first_object"
            - "<{unique_id}>"
```
Now I expect the generated values in the first and second object to be the same, as they refer to the same parameter, however for each a new uniqid has been generated.

Tested for version 2.2.2

**Comment by theofidry** [comment]
I certainly didn't expect it to be evaluated each time. @tshelburne is it the desired behaviour? I'm kinda surprise by it. Maybe I could fix it in 2.x but at the same time this feature has been around for a while and maybe some people are relying on it, so fixing it would be a BC break...

**Comment by tshelburne** [comment]
Hm, I definitely wouldn't expect this either. I think we should leave this ticket open with RFC, flag it in the docs somewhere, and then make sure it is happening as expected in 3.0. At this point, I doubt it's worth the BC break, but I don't have a great suggestion for @svandervlugt... The only idea I have in mind to avoid BC break is to add a flag to the parser that "eager evaluates" parameters. @theofidry Thoughts?

**Comment by theofidry** [comment]
Fine by me :)

**Comment by theofidry** [comment]
Closed by #689 and #688.
